homeassistant/components/conversation/default_agent.py

```diff
@@ -187,12 +187,22 @@ async def async_recognize(
             _LOGGER.warning("No intents were loaded for language: %s", language)
             return None
 
+        intent_context: dict[str, Any] | None = None
+
+        if user_input.device_id and (
+            device := dr.async_get(self.hass).async_get(user_input.device_id)
+        ):
+            intent_context = {"device_id": device.id}
+            if device.area_id:
+                intent_context["area"] = device.area_id
```

Comment:
I also think that this is not a good approach. It should be offered as a fallback to intent handlers, it shouldn't override and be presented as user provided info.

Reply:
As that will lead exactly to the breaking change in the PR description.

Reply:
I don't think I understand. The entire end-goal of this PR is to provide context to the recognizer, so that it can act locally, depending on the satellite that was used.

As far as I understood from @synesthesiam (I can't find the message rn), there won't ever be a priority system for sentences, so I don't know how you could guarantee that a fallback is a fallback and not the first option. My approach to writing sentences was to always have a single one match a given prompt. If a fallback mechanism is a possible solution, I am totally unaware of it.

Reply:
There are some sentences that will act differently if an area is defined. With this change, there will always be an area defined.

This means that asking "What is the temperature" will always be scoped to the area, instead of HA looking up the current temperature reported by the thermostat.

Reply:
So if we have a new parameter like `device_area`, we can allow an intent handler that matched a sentence decide to take `device_area` into account or not. With the current appraoch, the intent handler wouldn't know if the user specified the area or not.

Reply:
> With this change, there will always be an area defined.

...as long as the satellite has an assigned area, which may not always be the case.

> So if we have a new parameter like device_area, we can allow an intent handler that matched a sentence decide to take device_area into account or not.

Understood, i think. But I think you mean the conversation agent, not the intent handler, since the intent handler doesn't get the actual uttered sentence or context. I still think it's a better idea for intent handlers not to care how exactly the conversation agent came to the conclusion that the scope should be limited to an area or not.

I guess you could even expose such a user option at the agent level: should the agent limit scope to satellite area (when available) or not?